platform-sdk/swirlds-cli/src/main/java/com/swirlds/cli/utility/LogLine.java

```diff
@@ -0,0 +1,270 @@
+        logLevelOriginalString = logLineMatcher.group(5);
+        logLevel = LogLevel.valueOf(logLevelOriginalString);
+        whitespaces.add(logLineMatcher.group(6));
+        marker = logLineMatcher.group(7);
+        whitespaces.add(logLineMatcher.group(8));
+        threadName = logLineMatcher.group(9);
+        whitespaces.add(logLineMatcher.group(10));
+        className = logLineMatcher.group(11);
+        colonSpace = logLineMatcher.group(12);
+
+        final String remainderString = logLineMatcher.group(13);
+
+        switch (marker) {
+            case "PLATFORM_STATUS" -> remainderOfLine = new PlatformStatusLog(remainderString);
+            default -> remainderOfLine = new DefaultFormattableString(remainderString);
+        }
+    }
+
+    @NonNull
+    public Instant getTimestamp() {
+        return timestamp;
+    }
+
+    @NonNull
+    public String getLogNumber() {
+        return logNumber;
+    }
+
+    @NonNull
+    public LogLevel getLogLevel() {
+        return logLevel;
+    }
+
+    @NonNull
+    public String getMarker() {
+        return marker;
+    }
+
+    @NonNull
+    public String getThreadName() {
+        return threadName;
+    }
+
+    @NonNull
+    public String getClassName() {
+        return className;
+    }
+
+    @NonNull
+    public FormattableString getRemainderOfLine() {
+        return remainderOfLine;
+    }
+
+    /**
+     * Get the correct color for a given log level.
+     *
+     * @param logLevel the log level
+     * @return the color
+     */
+    @NonNull
+    private Color getLogLevelColor(@NonNull final LogLevel logLevel) {
+        return switch (logLevel) {
+            case TRACE, DEBUG, INFO -> HARMLESS_LOG_LEVEL_COLOR;
+            case WARN -> WARN_LOG_LEVEL_COLOR;
+                // all other log levels are critical
+            default -> ERROR_LOG_LEVEL_COLOR;
+        };
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public String getOriginalPlaintext() {
+        return originalLogString;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public String generateAnsiString() {
+        int whitespaceIndex = 0;
+
+        return colorizeStringAnsi(timestampOriginalString, TIMESTAMP_COLOR)
+                + whitespaces.get(whitespaceIndex++)
+                + colorizeStringAnsi(logNumber, LOG_NUMBER_COLOR)
+                + whitespaces.get(whitespaceIndex++)
+                + colorizeStringAnsi(logLevelOriginalString, getLogLevelColor(logLevel))
+                + whitespaces.get(whitespaceIndex++)
+                + colorizeStringAnsi(marker, LOG_MARKER_COLOR)
+                + whitespaces.get(whitespaceIndex++)
+                + colorizeStringAnsi(threadName, THREAD_NAME_COLOR)
+                + whitespaces.get(whitespaceIndex)
+                + colorizeStringAnsi(className, CLASS_NAME_COLOR)
+                + colonSpace
+                + remainderOfLine.generateAnsiString();
+    }
```

Comment:
If you use `TextEffect`, you can use `TextEffect.apply(string)`.

Reply:
Done